latex.bzl

```diff
@@ -1,26 +1,41 @@
-def _latex_pdf_impl(ctx):
+LatexOutputInfo = provider(fields = ['format', 'file'])
+
+def _latex_impl(ctx):
     toolchain = ctx.toolchains["@bazel_latex//:latex_toolchain_type"].latexinfo
     custom_dependencies = []
     for srcs in ctx.attr.srcs:
         for file in srcs.files.to_list():
             if file.dirname not in custom_dependencies:
                 custom_dependencies.append(file.dirname)
     custom_dependencies = ','.join(custom_dependencies)
+
+    ext = ".pdf"
+    for value in ctx.attr.cmd_flags:
```

Comment:
I agree. I'll update. But I find it hard to choose when it can also be sent as a flag. Via cmd_flags.

And so it becomes unfortunate that you could have format="pdf" by default;
and send in --latex-args=--output-format=dvi. Which one takes precedence?

Typical user error maybe, but it might not be obvious. 

So I started with one solution and changed halfway 😂 it seems.

Reply:
That's definitely a good point. I would say user error for now - eventually, I would very much love to get rid of `cmd_flags` entirely. I'd be curious to know what people are using it for - and expose those use-cases through specific arguments, rather than relying on the underlying tool through `cmd_flags`.

As a stopgap for now, one solution might be to look into `cmd_flags` - if a user passes `output-format` in there, call `fail`. While not super clean, that leads the user towards the "proper" and better supported way to do that.

Reply:
Yeah, that's a good idea.
Regarding remove the flags, that's also one of things I'd have a hard time to decide on.

In one way you simplify for the user, on the other you might build in a maintenance burden of keeping up to date with a new version of tool. For which you might get any gain except for an increasingly bloated interface. ( not to mention if we support multiple tools in the future, lualatex, xetex)

That said, as you did, I guess the users use case would be guiding.

One can always write an additional rule later 😊

I'm fine with any solution.
I think I have narrow use case and do not use latex so much in my day to day work. So my votes shouldn't be assigned much weight 😊

Reply:
There's definitely a point to be made about leaving `cmd_flags`, it's obviously a balance between flexibility and comfort of use. @patbro might have an opinion on that as well.

For this specific instance, given that the format impacts the name of the output file, I think we _have_ to have the format as a rule argument to declare the output.